api/v1alpha1/iochaos_webhook.go

```diff
@@ -126,3 +127,21 @@ func (in *IoChaosSpec) validatePercent(percentField *field.Path) field.ErrorList
 
 	return allErrs
 }
+
+func (in *IoChaosSpec) validateMistake(mistakeField *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	mistake := in.Mistake
+	if in.Mistake == nil {
+		return allErrs
+	}
+	allErrs = append(allErrs, in.validatePercent(mistakeField.Child("percent"))...)
+	// Useless since they're unsigned values. But keep it here for further tweaks.
+	if mistake.MaxLength <= 0 {
+		allErrs = append(allErrs, field.Invalid(mistakeField.Child("maxLength"), mistake.MaxLength, "maxLength should be greater than 0"))
+	}
+	if mistake.MaxOccurrences <= 0 {
+		allErrs = append(allErrs, field.Invalid(mistakeField.Child("maxOccurrences"), mistake.MaxOccurrences, "maxOccurrences should be greater than 0"))
+	}
```

Comment:
I prefer to put validation in the definition of that CRD with `// +kubebuilder:validation:<something>`.

Reference: https://book.kubebuilder.io/reference/markers/crd-validation.html

Reply:
We have used these markers in many places, you could find them by searching `+kubebuilder:validation`. :)

Reply:
Cool!

Reply:
Yes. 😁